javaslang/src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -133,24 +133,21 @@ private TreeMap(RedBlackTree<Tuple2<K, V>> entries) {
     /**
      * Creates a TreeMap of the given list of key-value pairs.
      *
+     * @deprecated Should not be used any more because unsafe
+     *
      * @param pairs A list of key-value pairs
      * @param <K>   The key type
      * @param <V>   The value type
      * @return A new Map containing the given entries
      */
+    @Deprecated
     @SuppressWarnings("unchecked")
     public static <K extends Comparable<? super K>, V> TreeMap<K, V> of(Object... pairs) {
-        return of((Comparator<? super K> & Serializable) K::compareTo, pairs);
-    }
-
-    @SuppressWarnings("unchecked")
-    public static <K, V> TreeMap<K, V> of(Comparator<? super K> keyComparator, Object... pairs) {
```

Comment:
@danieldietrich please check this code.
This method removed instead of make them deprecated. Compiler says "possible ambiguous with of(keyComparator, k1, v1, ..., k10, v10)". The problem observed only with TreeMap and only in this line of code. I have no idea whats wrong